subprojects/base-services/src/test/groovy/org/gradle/internal/concurrent/DefaultExecutorFactoryTest.groovy

```diff
@@ -122,23 +122,28 @@ class DefaultExecutorFactoryTest extends ConcurrentSpec {
     }
 
     def stopThrowsExceptionOnTimeout() {
+        def timeoutMs = 100
         def action = {
             thread.block()
```

Comment:
Per our conversation, let's use an indefinite interruptible block here (e.g. CountDownLatch.await() or something similar)?  This _should_ avoid a race condition between when we start the action and we try to stop it.